napari/layers/_vectors_layer/model.py

```diff
@@ -358,6 +369,39 @@ def _refresh(self):
         self._need_display_update = True
         self._update()
 
+    def _generate_meshes(self, vectors, width):
+        """Generates list of mesh vertices and triangles from a list of vectors
+
+        Parameters
+        ----------
+        vectors : np.ndarray
+            Nx2 array where each pair of vertices corresponds to an independent
+            line segment
+        width : float
+            width of the line to be drawn
+
+        Returns
+        ----------
+        vertices : np.ndarray
+            2Nx2 array of vertices of all triangles for the lines
+        triangles : np.ndarray
+            Nx3 array of vertex indices that form the mesh triangles
+        """
+        centers = np.array([vectors[i//2] for i in range(2*len(vectors))])
+        offests = np.array([segment_normal(vectors[2*i], vectors[2*i+1]) for i
+                            in range(len(vectors)//2)])
+        offests = offests[[i//4 for i in range(4*len(offests))]]
+        signs = np.ones((len(offests), 2))
+        signs[::2] = -1
+        offests = offests*signs
```

Comment:
offests -> offsets

@sofroniewn I thought in your shapes PR you added some machinery for drawing lines — wouldn't it be better to reuse that machinery here? Or is there a reason why this is different?

Reply:
so the only part that is in common right now is the  `segment_normal` function. I'm going to add that to our model general utils.

Otherwise the actual mesh calculation code here is much simpler as I know that each line is simply a single line segment with only two endpoints. I don't need to worry about any of the bevel join / miter join stuff like I have to do in the other PR if we have lines with multiple line segments